server/apps/publish/content/common.py

```diff
@@ -86,61 +86,13 @@ class BasePublishService(BaseService):
-        if original.get('flags', {}).get('marked_for_not_publication', False):
-            raise SuperdeskApiError.badRequestError('Cannot publish an item which is marked as Not for Publication')
-
-    def raise_if_invalid_state_transition(self, original):
-        if not is_workflow_state_transition_valid(self.publish_type, original[ITEM_STATE]):
-            error_message = "Can't {} as item state is {}" if original[ITEM_TYPE] == CONTENT_TYPE.TEXT else \
-                "Can't {} as either package state or one of the items state is {}"
-            raise InvalidStateTransitionError(error_message.format(self.publish_type, original[ITEM_STATE]))
-
     def on_update(self, updates, original):
-        self.raise_if_not_marked_for_publication(original)
-        self.raise_if_invalid_state_transition(original)
-
-        updated = original.copy()
-        updated.update(updates)
-
-        takes_package = self.takes_package_service.get_take_package(original)
-
-        if self.publish_type == 'publish':
-            # validate if take can be published
-            if takes_package and not self.takes_package_service.can_publish_take(
-                    takes_package, updates.get(SEQUENCE, original.get(SEQUENCE, 1))):
-                raise PublishQueueError.previous_take_not_published_error(
-                    Exception("Previous takes are not published."))
-
-            validate_schedule(updated.get('publish_schedule'), takes_package.get(SEQUENCE, 1) if takes_package else 1)
-
-            if original[ITEM_TYPE] != CONTENT_TYPE.COMPOSITE and updates.get(EMBARGO):
-                get_resource_service(ARCHIVE).validate_embargo(updated)
-
-        if self.publish_type in ['correct', 'kill']:
-            if updates.get(EMBARGO):
-                raise SuperdeskApiError.badRequestError("Embargo can't be set after publishing")
-
-            if updates.get('dateline'):
-                raise SuperdeskApiError.badRequestError("Dateline can't be modified after publishing")
-
-        validate_item = {'act': self.publish_type, 'type': original['type'], 'validate': updated}
-        validation_errors = get_resource_service('validate').post([validate_item])
-        if validation_errors[0]:
-            raise ValidationError(validation_errors)
-
-        # validate the package if it is one
-        package_validation_errors = []
-        self._validate_package_contents(original, takes_package, package_validation_errors)
-        if len(package_validation_errors) > 0:
-            raise ValidationError(package_validation_errors)
-
+        updates['digital_item_id'] = self._get_digital_id_for_package_item(original)
+        self._validate(original, updates)
         self._set_updates(original, updates, updates.get(config.LAST_UPDATED, utcnow()))
-        updates[ITEM_OPERATION] = ITEM_PUBLISH
-        convert_task_attributes_to_objectId(updates)
+        convert_task_attributes_to_objectId(updates) # ???
```

Comment:
Problem is how the task attributes are stored. It is stored as string or `ObjectId` and for mongo queries to work we need all as `ObjectId`.  Other option might be to have schema for the `task`?